python/cugraph_service/cugraph_service_server/cugraph_handler.py

```diff
@@ -677,20 +679,100 @@ def uniform_neighbor_sample(
             )
 
         try:
-            return call_algo(
+            uns_result = call_algo(
                 uniform_neighbor_sample,
                 G,
                 start_list=start_list,
                 fanout_vals=fanout_vals,
                 with_replacement=with_replacement,
             )
+            if (result_host is not None) or (result_port is not None):
+                if (result_host is None) or (result_port is None):
+                    raise ValueError(
+                        "both result_host and result_port must "
+                        "be set if either is set. Got: "
+                        f"{result_host=}, {result_port=}"
+                    )
+                asyncio.run(
+                    self.__ucx_send_results(
+                        result_host,
+                        result_port,
+                        uns_result.sources,
+                        uns_result.destinations,
+                        uns_result.indices,
```

Comment:
Curious on how would we transfer a dict of indices: {type: ['sources' ,'destinations', 'indices'} something that looks like that

Reply:
@VibhuJawa and I discussed this offline. We'll need the ability to have extension APIs return various numbers of arrays, dtypes, etc. which are user-defined, so we'll likely allow extension authors to specify meta-data about the arrays being returned and used to reconstruct the correct number of arrays and dtypes client-side.  This will be part of a future PR.